mlflow/models/evaluation/base.py

```diff
@@ -219,6 +236,24 @@ def __init__(self, data, labels, name=None, path=None):
         self.path = path
         self._hash = None
 
+        if isinstance(self.data, np.ndarray):
+            num_features = self.data.shape[1]
+            if feature_names is not None:
+                feature_names = list(feature_names)
+                if num_features != len(feature_names):
+                    raise ValueError('feature name list must be the same length with feature data.')
+                self._feature_names = feature_names
+            else:
+                self._feature_names = [f'f_{i}' for i in range(num_features)]
```

Comment:
f'f_{i}' -> f'f_{i}'.zfill(int(math.log10(len(num_features))) to pad 0s to the left for easy sorting.

Reply:
seems it should be `f'f_{str(i).zfill(math.ceil((math.log10(num_features))))}'`